viewer/packages/cad-model/src/wrappers/CadMeshManager.ts

```diff
@@ -45,44 +45,34 @@ export class CadMeshManager {
     const allTreeIndices = new Set<number>();
 
     parsedMeshGeometries.forEach(geometryData => {
-      if (geometryData.type === RevealGeometryCollectionType.TriangleMesh) {
-        this.createMeshFromGeometry(
-          group,
-          geometryData.geometryBuffer,
-          materials.triangleMesh,
-          geometryData.wholeSectorBoundingBox
-        );
+      const isTriangleMesh = geometryData.type === RevealGeometryCollectionType.TriangleMesh;
+      const isTexturedTriangleMesh = geometryData.type === RevealGeometryCollectionType.TexturedTriangleMesh;
 
-        // Collect tree indices for mapping
-        const treeIndices = this.createTreeIndexSet(geometryData.geometryBuffer);
-        for (const treeIndex of treeIndices.keys()) {
-          allTreeIndices.add(treeIndex);
-        }
-      } else if (geometryData.type === RevealGeometryCollectionType.TexturedTriangleMesh) {
-        if (geometryData.texture) {
-          const texturedMaterial = this._materialManager.addTexturedMeshMaterial(
+      if (isTriangleMesh || isTexturedTriangleMesh) {
+        let material: RawShaderMaterial;
+
+        if (isTexturedTriangleMesh && geometryData.texture) {
+          material = this._materialManager.addTexturedMeshMaterial(
             this._modelIdentifier,
             sectorId,
             geometryData.texture
           );
-
-          this.createMeshFromGeometry(
-            group,
-            geometryData.geometryBuffer,
-            texturedMaterial,
-            geometryData.wholeSectorBoundingBox
-          );
-
-          // Collect tree indices for mapping
-          const treeIndices = this.createTreeIndexSet(geometryData.geometryBuffer);
-          for (const treeIndex of treeIndices.keys()) {
-            allTreeIndices.add(treeIndex);
-          }
-        } else {
+        } else if (isTexturedTriangleMesh && !geometryData.texture) {
```

Comment:
In my opinion, I think it's fine to just not check for this case, (and then we could just have a ternary operator rather than an if-else statement), but we can keep it if you want.